python/lsst/daf/butler/core/_butlerUri.py

```diff
@@ -434,12 +434,25 @@ def getExtension(self) -> str:
         -------
         ext : `str`
             The file extension (including the ``.``). Can be empty string
-            if there is no file extension. Will return all file extensions
-            as a single extension such that ``file.fits.gz`` will return
-            a value of ``.fits.gz``.
+            if there is no file extension. Usually returns only the last
+            file extension unless there is a special extension modifier
+            indicating file compression, in which case the combined
+            extension (e.g. ``.fits.gz``) will be returned.
         """
+        special = {".gz", ".bz2", ".xz"}
```

Comment:
I think I see that `.fz` is not "special" because it is handled natively by the fits formatter.  Which is actually good, because then we can support `fits.fz` or `.fz` which I think are both acceptable.

Reply:
Yes, at the moment even though `.fits.fz` is listed as a supported extension in the formatter it's irrelevant because the `.fz` on its own is going to be enough to match.